src/forms/DateTimePicker.jsx

```diff
@@ -87,26 +90,62 @@ class DateTimePicker extends React.Component {
 
 		datetime.setFullYear(newDate.getFullYear(), newDate.getMonth(), newDate.getDate());
 		this.setState({ datetime });
+
+		this.dateComponent.setState({
+			value: datetime
+		});
+
+		// update flatpickr on update
+		this.dateComponent.updateFlatpickr();
+
+		if (this.props.onChangeCallback) {
+			this.props.onChangeCallback(datetime);
+		}
 	}
 
 	/**
 	* @function parseTimeFromDateTime
 	* @description gets the time string part of our datetime state
+	* @param Date datetime - datetime object to be parsed
 	* @return String
 	*/
 	parseTimeFromDateTime(datetime) {
 		return (datetime.toTimeString().split(' ')[0]).split(':', 2).join(':'); // TODO localize toLocaleTimeString ?
 	}
 
 	/**
+	* @function parseHoursAndMinutesFromTimeString
+	* @description takes a time string and returns an array of hours and minutes
+	* @param String timeString a 24-hour time string, eg `18:45`
+	* @return Array [0] - the hours, [1] - the minutes
+	*/
+	parseHoursAndMinutesFromTimeString(timeString) {
+		return timeString.split(':'); // TODO localization?
+	}
+
+
+	/**
+	* @function parseNewTimeAsDate
+	* @description takes a time string in 24hr time (e.g. `18:35`)
+	* and converts it to a Date object with the current state's date and timezone
+	* @return Date new date with current date and the given time
+	*/
+	parseNewTimeAsDate(timeStr) {
+		const datetime = this.state.datetime;
+		const parsedTime = this.parseHoursAndMinutesFromTimeString(timeStr);
+
+		datetime.setHours(parsedTime[0]);
+		datetime.setMinutes(parsedTime[1]);
```

Comment:
check out [the signature for `setHours`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setHours) - you can `setMinutes` by passing the minutes as a second arg.